### PR TITLE
Git integration, readme fixes, style changes

### PR DESCRIPTION
**Pull request by jonluca**
I added the git integration, fixed up some parts of the readme. and fixed some style issues like mismatched brackets and such.

let me know if you have any questions?

### CODE REVIEW

File: done.js [hunk 1]
```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
...
-	} else {
-		displayError("No tasks added!!");
-	}
+}
 
+// Initalizes git integration
+function gitInit() {
+    // If we're not currently in  a git repository, exit
+    if (!fs.existsSync('.git')) {
+        displayError("Not currently in a git repository!");
```
Comment (v3rse) [inline done.js:210]: It would be cool if you created a git repo by running `git init`. What do you think?

File: done.js [hunk 2]
```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
...
+//checks if the file exists.
+//If it does, it just calls back.
+//If it doesn't, then the file is created.
+// credit to https://stackoverflow.com/questions/12899061/creating-a-file-only-if-it-doesnt-exist-in-node-js
+function checkForFile(fileName, callback) {
+    fs.exists(fileName, function (exists) {
+        if (exists) {
+            callback();
+        } else {
+            fs.writeFile(fileName, {flag: 'wx'}, function (err, data) {
```
Comment (v3rse) [inline done.js:319]: I believe the options i.e. `{flag: 'wx'}` are supposed to be the third argument. The second argument is supposed to be the data written to the file. Check out the docs [here](http://devdocs.io/node/fs#fs_fs_writefile_file_data_options_callback). This currently writes `[object Object]` to the file.

File: done.js [hunk 3]
```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
...
+            var shaRegex = /^\[(.*) (.*)\]/gm;
+            var arr = shaRegex.exec(stdout);
+            const commitSha = arr[2];
+            //Alert the git commit has been created
+            console.log("Committed to git with SHA " + commitSha);
+            data.uncompleted[task].commit = commitSha;
+
+            //save data
+            setData(data);
+            console.log("Saved commit to task " + task + 1);
```
Comment (v3rse) [inline done.js:266]: I might be cool to call `check(task)` here to check the task after the commit is added.

**Review by v3rse — CHANGES_REQUESTED** [review]
@jonluca Sorry for taking so long with this. Was swamped on a work project. Awesome work!! I just noticed a few things which I commented on. Again thanks.

File: done.js [hunk 4]
```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
...
 }
 
 
-
 //Displays usage
 function usage() {
-	console.log("Usage: done [add|check|delete|help|clear [all|done]|list [all|done]] [task]");
-	console.log("`task` is only a string when using `add` and a number\nfor all other commands.");
-	console.log("Using the `done` without arguments lists all tasks");
+    console.log("Usage: done [add|check|delete|help|clear [all|done]|list [all|done]] [task]");
```
Comment (v3rse) [inline done.js]: Forgot to add this as well. We will need to add the `git` commands to the usage function

**Comment by v3rse** [comment]
@jonluca Forgot to add this as well. We will need to add the git commands to the usage function.

**Comment by jonluca** [comment]
Closing this as I don't have the cycles right now. Thanks!

**Pull request closed without merging**